src/main/java/com/sap/oss/phosphor/fosstars/model/score/oss/ArtifactReleaseHistoryScore.java

```diff
@@ -139,6 +153,28 @@ static VersionStats calculateVersionStats(Collection<VersionInfo> versionInfoCol
     return versionInfo;
   }
 
+  /**
+   * Filter the artifact versions matching with the major version of the given
+   * {@link ArtifactVersion}. Return the filtered list if size > 1.
+   * 
+   * @param artifactVersions {@link ArtifactVersions} to be filtered.
+   * @param artifactVersion {@link ArtifactVersion} provided by the user.
+   * @return A collection of {@link ArtifactVersion}.
+   */
+  private static Collection<ArtifactVersion> filter(Value<ArtifactVersions> artifactVersions,
+      Value<ArtifactVersion> artifactVersion) {
+    if (!artifactVersion.isUnknown() && artifactVersion.get().hasValidSemanticVersion()) {
+      ArtifactVersions filteredArtifactVersions =
+          artifactVersions.get().filterArtifactsByMajorVersion(
+              artifactVersion.get().getSemanticVersion().get().toString());
```

Comment:
`filterArtifactsByMajorVersion()` takes a string and then parses it to a semantic version. Instead, it can accept `SemanticVersion`.

```suggestion
          artifactVersions.get().filterArtifactsByMajorVersion(artifactVersion.get().semanticVersion());
```